asset/src/__lib/accumulator.ts

```diff
@@ -22,7 +22,7 @@ export default class Accumulator {
     flush(): DataEntity[] {
         const results = this.records;
         this.records = [];
-        return results.slice();
+        return results;
```

Comment:
this is a wasted operation, there is no reason to try to keep it immutable as we are directly dereferencing/deleting the old data on purpose.

Reply:
For a few reasons, it's not expensive, it makes the intent clear that you are not passing along the reference and avoid any accidental mutations, it removes any deferenced values (sparse arrays) and it is also better for the garbage collector because those reasons

Reply:
I would not call an O(N) time complexity cheap, there are no accidental mutations as the `this.records` is a brand new array that is seperate from results. This is such a minor thing to hold up this PR so I am going to make the change so others can keep working

Reply:
I mean we are talking about an upfront cost of roughly a millisecond for an array with a million values but it allows the garbage collector to work more efficiently (meaning it may speed things up instead of slowing things down).